devito/symbolics/extended_sympy.py

```diff
@@ -537,6 +537,28 @@ class DOUBLEP(CastStar):
 MAX = Function('MAX')
 MIN = Function('MIN')
 
+
+def rfunc(func, item, *args):
+    """
+    A utility function that recursively generates 'func' nested relations.
+
+    Examples
+    ----------
+    >> rfunc(min, [a, b, c, d])
+    MIN(a, MIN(b, MIN(c, d)))
+
+    >> rfunc(max, [a, b, c, d])
+    MAX(a, MAX(b, MAX(c, d)))
+    """
+
+    try:
```

Comment:
this should probably be an assert (https://stackoverflow.com/questions/4333371/when-to-use-assert-and-when-to-use-try-catch)